     @Override
-    protected void onPlace(IBlockData iblockdata, World world, BlockPosition blockposition, IBlockData iblockdata1, boolean flag) {
+    protected void onPlace(IBlockData iblockdata, World world, BlockPosition blockposition, IBlockData iblockdata1, boolean flag, ItemActionContext context) { // CraftBukkit - context
         if (!iblockdata1.is(iblockdata.getBlock())) {
             if (inPortalDimension(world)) {
                 Optional<BlockPortalShape> optional = BlockPortalShape.findEmptyPortalShape(world, blockposition, EnumDirection.EnumAxis.X);
 
                 if (optional.isPresent()) {
-                    ((BlockPortalShape) optional.get()).createPortalBlocks();
+                    ((BlockPortalShape) optional.get()).createPortalBlocks((context == null) ? null : context.getPlayer()); // CraftBukkit - player
                     return;
                 }
             }
 
             if (!iblockdata.canSurvive(world, blockposition)) {
-                world.removeBlock(blockposition, false);
+                fireExtinguished(world, blockposition); // CraftBukkit - fuel block broke
             }
 
         }
     }
 
     private static boolean inPortalDimension(World world) {
-        return world.dimension() == World.OVERWORLD || world.dimension() == World.NETHER;
+        return world.getTypeKey() == net.minecraft.world.level.dimension.WorldDimension.OVERWORLD || world.getTypeKey() == net.minecraft.world.level.dimension.WorldDimension.NETHER; // CraftBukkit - getTypeKey()
     }
 
     @Override
@@ -203,4 +214,12 @@
             }
         }
     }
}